File: nodejs-tools/nodejs-compiler-tools/formatter-xml/src/format_text_node.ts
import { XmlTextNode } from '@vighnesh153/parser-xml';
import { buildIndentationSpace } from './build_indentation_space.ts';

type FormatTextNodeOptions = {
  textNode: XmlTextNode;
  indentationLevel: number;
  indentation: number;
};

export function formatTextNode({ textNode, indentationLevel, indentation }: FormatTextNodeOptions): string {
  return buildIndentationSpace({ indentationLevel, indentation }) + textNode.text.tokenLiteral.trim();
}

File: tools-jvm/XmlFormatter/src/main/kotlin/dev/vighnesh153/xml/formatter/fmt/FormatTextNode.kt
package dev.vighnesh153.xml.formatter.fmt

import dev.vighnesh153.xml.formatter.ast.XmlTextNode
import dev.vighnesh153.xml.formatter.ast.buildIndentation

internal fun XmlTextNode.format(
    indentationLevel: Int,
    indentation: Int,
): String = "${
    buildIndentation(
        indentationLevel = indentationLevel,
        indentation = indentation
    )
}${text.tokenLiteral.trim()}"
